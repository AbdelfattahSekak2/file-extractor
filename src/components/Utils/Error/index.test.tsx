import { mount, ReactWrapper, shallow, ShallowWrapper } from "enzyme";
import React from "react";

import ErrorMessage from ".";

describe("ErrorMessage component", () => {
  const clearError = jest.fn();
  let wrapper: ShallowWrapper | ReactWrapper;

  beforeEach(() => {
    wrapper = shallow(<ErrorMessage error="error" clearError={clearError} />);
  });

  it("should render correctly", () => {
    expect(wrapper).not.toBeNull();
    expect(wrapper).toMatchSnapshot();
  });

  it("should open modal correctly when error is updated", () => {
    wrapper = mount(<ErrorMessage error={null} clearError={clearError} />);
    expect(wrapper.find("Modal").prop("open")).toEqual(false);

    wrapper = mount(<ErrorMessage error="error" clearError={clearError} />);
    expect(wrapper.find("Modal").prop("open")).toEqual(true);
  });

  it("should handle clear state correctly", () => {
    wrapper.find("Button").simulate("click");
    expect(wrapper.find("Modal").prop("open")).toEqual(false);
    expect(clearError).toHaveBeenCalled();
  });

  it("should handle on close state correctly", () => {
    // @ts-ignore
    wrapper.find("Modal").props().onClose();
    expect(wrapper.find("Modal").prop("open")).toBeFalsy();
  });
});
