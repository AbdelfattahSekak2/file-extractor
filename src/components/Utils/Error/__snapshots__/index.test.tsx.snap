// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage component should render correctly 1`] = `
<Modal
  centered={true}
  className="error-message"
  closeOnDimmerClick={true}
  closeOnDocumentClick={false}
  dimmer={true}
  eventPool="Modal"
  onClose={[Function]}
  open={false}
>
  <ModalHeader>
    <Icon
      as="i"
      name="frown"
      size="big"
    />
    error
  </ModalHeader>
  <ModalContent>
    <Button
      as="button"
      onClick={[Function]}
    >
      Try again
    </Button>
  </ModalContent>
</Modal>
`;
